BLUESPAWN-client/src/user/CLI.cpp

```diff
@@ -7,82 +7,132 @@
+	Print(L" " + prompt);
 
+	int i = 0;
 	for (auto str : options) {
-		stream << i << L". " << str << endl;
-		i++;
+		i += 1;
+		Print(std::to_wstring(i), MessageColor::CYAN, false);
+		Print(L". " + str);
 	}
-	wcout << stream.str() << endl;
-	
-	int userIn;
-	while (true) {
-		wcin >> userIn;
-		if (userIn > 0 && userIn <= options.size()) {
-			return options[userIn-1];
-		}
-		else {
-			wcout << L"Please Enter a valid number between 1 and " << options.size() <<endl;
+	Print(L"Please enter a number 1 through " + std::to_wstring(i) + L" to continue. ", MessageColor::LIGHTGRAY, false);
+
+	while(true){
+		int userIn = 0;
+		std::wstringstream stream{};
+		auto input = GetInput(dwMaximumDelay);
+		if(!input){
+			return {};
+		} else {
+			stream << *input;
+			stream >> userIn;
+			if(stream.good() && userIn > 0 && userIn <= i){
+				return options[userIn - 1];
+			} else {
+				Print(L"Please enter a number 1 through " + std::to_wstring(i) + L" to continue. ", MessageColor::LIGHTGRAY, false);
+			}
 		}
-	}
+	};
+
 	return L"";
 }
 
-void CLI::InformUser(const std::wstring& information) const {
-	SetConsoleColor(MessageColor::BLUE);
-	wcout << L"[*] ";
-	SetConsoleColor(MessageColor::WHITE);
-
-	wcout << information << endl;
+void CLI::InformUser(const std::wstring& information, ImportanceLevel level) const {
+	auto mutex = AcquireMutex(hMutex);
+	Print(INFORM_ID, ID_COLOR, false);
+	Print(descriptions[static_cast<DWORD>(level)], colors[static_cast<DWORD>(level)], false);
+	Print(L" " + information);
 }
-bool CLI::AlertUser(const std::wstring& information, DWORD dwMaximumDelay) const { 
-	SetConsoleColor(MessageColor::BLUE);
-	wcout << L"[!] ";
-	SetConsoleColor(MessageColor::WHITE);
-	wcout << information << endl;
-	wcin.ignore(numeric_limits<streamsize>::max(),'\n');
-	return true;
+bool CLI::AlertUser(const std::wstring& information, DWORD dwMaximumDelay, ImportanceLevel level) const {
+	auto mutex = AcquireMutex(hMutex);
+	Print(ALERT_ID, ID_COLOR, false);
+	Print(descriptions[static_cast<DWORD>(level)], colors[static_cast<DWORD>(level)], false);
+	Print(L" " + information);
+	Print(L"Press enter to continue. ", MessageColor::LIGHTGRAY, false);
+	return GetInput(dwMaximumDelay).has_value();
 }
-const set<wstring> affirmativeOptions = { L"yes", L"y"};
-const set<wstring> negativeOptions = { L"no", L"n" };
-const set<wstring> cancelOptions = { L"cancel" };
-DWORD CLI::GetUserConfirm(const std::wstring& prompt, DWORD dwMaximumDelay) const {
-	wstring result;
-	wcout << prompt << endl;
-	wcin >> result;
-	result = ToLowerCase(result);
-	if (affirmativeOptions.find(result)!=affirmativeOptions.end()) {
-		return 1;
-	}
-	else if (negativeOptions.find(result) != negativeOptions.end()) {
-		return 0;
-	}
-	else {
-		return -1;
+
+DWORD CLI::GetUserConfirm(const std::wstring& prompt, DWORD dwMaximumDelay, ImportanceLevel level) const {
+	auto mutex = AcquireMutex(hMutex);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'mutex' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5011005)